CoreData/VICoreDataManager/SKZPlistMapExtensions.m

```diff
@@ -0,0 +1,162 @@
+#import "VIManagedObject.h"
+#import "VICoreDataManagerSKZ.h"
+#import "VIManagedObjectMapperSKZ.h"
+
+NSString *SKZArchivedClassName = @"_isa";
+
+
+@implementation NSObject (SKZPlistMapExtensions)
+
+- (id)createObjectForKey:(NSString *)key owner:(id)owner context:(NSManagedObjectContext *)context
+{
+    return self;
+}
+
+@end
+
+@implementation NSArray (SKZPlistMapExtensions)
+
+
+- (id)createObjectForKey:(nonnull NSString *)key owner:(id)owner context:(NSManagedObjectContext *)context
+{
+    NSUInteger count = [self count];
+    id members[count];
+    NSUInteger i = 0;
+    
+    for ( id plistObj in self )
+    {
+        members[i] = [plistObj createObjectForKey:key owner:owner context:context];
+        i++;
+    }
+    
+    NSSet *memberSet = [NSSet setWithObjects:members count:count];
+    return memberSet;
+}
+
+@end
+
+@implementation NSDictionary (SKZPlistMapExtensions)
+
+- (id)createObjectForKey:(NSString *)ownerKey className:(NSString *)className owner:(id)owner context:(NSManagedObjectContext *)context
+{
+    id managedRootObject = nil;
+    
+    if ( className == Nil )
+    {
+        return self;     // No class specifier found, return as raw dict
+    }
+    
+    Class managedObjectClass = NSClassFromString(className);
+    
+    if ( managedObjectClass == Nil )
+    {
+        return self;    // No class found, return as raw dict
+    }
+    
+    VICoreDataManagerSKZ *cdm = [VICoreDataManagerSKZ sharedInstance];
+    context = [cdm safeContext:context];
+    
+    VIManagedObjectMapperSKZ *mapper = [cdm mapperForClass:managedObjectClass];
+    
+    id comparisonValue = [self objectForKey:mapper.foreignUniqueComparisonKey];
+    
+    // check for existing object with matching key
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:className];
+    fetchRequest.predicate = [NSPredicate predicateWithFormat:@"(%K == %@)", mapper.uniqueComparisonKey, comparisonValue];
+    
+    NSError *error;
+    NSArray *matchingObjects = [context executeFetchRequest:fetchRequest error:&error];
+    
+    if (error)
+    {
+        CDLog(@"%s Fetch Request Error\n%@",__PRETTY_FUNCTION__ ,[error localizedDescription]);
+    }
+
+    NSUInteger matchingObjectsCount = matchingObjects.count;
+    
+    if ( matchingObjectsCount > 0 )
+    {
+        NSAssert(matchingObjectsCount < 2, @"UNIQUE IDENTIFIER IS NOT UNIQUE. MORE THAN ONE MATCHING OBJECT FOUND");
+        managedRootObject = matchingObjects[0];
+    }
+    else
+    {
+        // no match found, so create a new one
+        managedRootObject = [cdm managedObjectOfClass:managedObjectClass inContext:context];
+    }
+    
+    // assign properties and subobjects
+    for ( NSString * inputKeyPath in self )
+    {
+        id plistValue = [self objectForKey:inputKeyPath];
+        VIManagedObjectMapSKZ *map = [mapper mapForInputKeyPath:inputKeyPath];
+        NSString *coreDataKey = map.coreDataKey;
+        
+        if ( coreDataKey == nil )
```

Comment:
So since the relationships aren't specified in the mapper, we could use the same key in the dict we send as the relationship property:

```
NSString *propKey = map.coreDataKey ?: inputKeypath;
if ( propKey == nil ) {
    continue;
}
```